dspace/modules/journal-submit/src/main/java/org/datadryad/submission/DryadEmailSubmission.java

```diff
@@ -70,162 +68,98 @@
 
     // Maps not concurrent but we only access, don't write to except at init()
     // map of Journal Codes to Journals
-    private static Map<String, PartnerJournal> myJournals; 
-    
+    private static Map<String, PartnerJournal> myJournals;
+
     // map of Journal Names to Journal Codes (in case code is not preent in submission)
     private static Map<String, String> myJournalNames;
 
+    // Timer for scheduled harvesting of emails
+    private Timer myEmailHarvester;
+
+    private static DryadGmailService dryadGmailService;
+
     /**
+     * UPDATE: GET only works for authorization and testing Gmail API.
      * Handles the HTTP <code>GET</code> method by informing the caller that
      * <code>GET</code> is not supported, only <code>POST</code>.
-     * 
-     * @param aRequest A servlet request
+     *
+     * @param aRequest  A servlet request
      * @param aResponse A servlet response
      * @throws ServletException If a servlet-specific error occurs
-     * @throws IOException If an I/O error occurs
+     * @throws IOException      If an I/O error occurs
      */
     @Override
     protected void doGet(HttpServletRequest aRequest,
-            HttpServletResponse aResponse) throws ServletException, IOException {
-        aResponse.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED,
-        "GET is not supported, you must POST to this service");
+                         HttpServletResponse aResponse) throws ServletException, IOException {
+        String requestURI = aRequest.getRequestURI();
+        if (requestURI.contains("authorize")) {
+            String queryString = aRequest.getQueryString();
+            if (aRequest.getQueryString() == null) {
+                // If we've never gotten a credential from here before, do this.
+                String urlString = dryadGmailService.getAuthorizationURLString();
+                LOGGER.info("userID "+dryadGmailService.getMyUserID());
+                aResponse.sendRedirect(urlString);
+            }
+            else if (queryString.contains("code=")) {
+                String code = queryString.substring(queryString.indexOf("=")+1);
+                LOGGER.info("authorizing with code "+ code);
+                // Generate Credential using retrieved code.
+                dryadGmailService.authorize(code);
+            }
+            return;
+        } else if (requestURI.contains("test")) {
+            try {
+                dryadGmailService = new DryadGmailService();
+                ArrayList<String> labels = new ArrayList<String>();
+                labels.add("test_gmail_api");
+                List<Message> messages = dryadGmailService.retrieveMessagesWithLabels(labels);
```

Comment:
When the labels here don't exist in the gmail account, this method prints out every message to the log and does not send a response